src/daft-local-execution/src/sources/scan_task.rs

```diff
@@ -284,9 +286,89 @@ async fn stream_scan_task(
         }
         #[cfg(feature = "python")]
         StorageConfig::Python(_) => {
-            return Err(common_error::DaftError::TypeError(
-                "Streaming reads not supported for Python storage config".to_string(),
-            ));
```

Comment:
Isn't this codepath for the non-native reads? I'm assuming it's fine to drop support for non-native reads?